### ISSUE TITLE
Invalid memory access exception

### ISSUE THREAD

**Issue by BullyWiiPlaza** [dblock/oshi#221]
I'm running `Windows 10` and sometimes when the following code is executed...

```
SystemInfo systemInfo = new SystemInfo();
HardwareAbstractionLayer hardwareAbstractionLayer = systemInfo.getHardware();
CentralProcessor centralProcessor = hardwareAbstractionLayer.getProcessor(); // Exception here (non-deterministic)
```

...it gives me the following exception:
```
Exception in thread "main" java.lang.Error: Invalid memory access
	at com.sun.jna.Native._getPointer(Native Method)
	at com.sun.jna.Native.getPointer(Native.java:1962)
	at com.sun.jna.Pointer.getPointer(Pointer.java:663)
	at com.sun.jna.platform.win32.COM.COMInvoker._invokeNativeInt(COMInvoker.java:25)
	at com.sun.jna.platform.win32.COM.Unknown.Release(Unknown.java:70)
	at oshi.util.platform.windows.WmiUtil.queryWMI(WmiUtil.java:307)
	at oshi.util.platform.windows.WmiUtil.selectStringsFrom(WmiUtil.java:215)
	at oshi.hardware.platform.windows.WindowsCentralProcessor.<clinit>(WindowsCentralProcessor.java:82)
	at oshi.hardware.platform.windows.WindowsHardwareAbstractionLayer.getProcessor(WindowsHardwareAbstractionLayer.java:55)
```

What could be the cause of this? A bug?

**Comment by dbwiddis** [comment]
Yes, likely a bug but I'm not sure where.  

The error is occurring when the processor is initializing and matching tick values to up time to store its conversion factor; it's making a wmi query.

It looks like I may be doubly-releasing the enumerator in the (apparently nondeterministic) case that the query fails.  So I think there may be an easy fix for this symptom.  

Note that I'm developing 3.0 now, so the fix will be in 3.0-SNAPSHOT.  You may want to migrate to the 3.0-rc1 version of the code first (you may still see the bug there).

The bigger question is why the query is failing sometimes and if there's a way I can prevent that.

**Comment by dbwiddis** [comment]
Could you please run the following on your windows command line, several times, to see if anything unusual happens:
```
wmic PATH Win32_PerfRawData_Counters_ProcessorInformation GET Name,PercentIdleTime,PercentPrivilegedTime,PercentUserTime
```
By "unusual" I mean there should be a line that reads "_Total" and if it either doesn't return or returns errors, that is of interest.

**Comment by BullyWiiPlaza** [comment]
Running your command the first time took a few seconds of processing but then still returned the table. I then ran it about 50 times and the result returned very quickly every time but nothing unusual happened.

**Comment by dbwiddis** [comment]
OK, I've fixed the problem in 3.0-SNAPSHOT.  I removed the double-release so the bug shouldn't recur.  The code during initialization was actually unnecessary (I dug into more documentation to avoid the need for that calculation) so it's been removed.  And I've made sure that if a 0 array is returned (which would happen on failure) the program behaves nicely.
